text/0000-artifact-hub-integration.md

```diff
@@ -0,0 +1,339 @@
+
+At this point, ArtifactHub.io will be scanning the registry index using the new file structure outlined in the previous
+step. Instead of the CNB registry service scanning this as well, the registry service will populate its database based
+on calls to the ArtifactHub.io API.
+
+When this step is complete, publishers will be able to take full owership of their publishing process. The buildpacks
+registry index will continue to be read/write for 12 months, so publishers can continue to publish to the buildpacks
+registry.
+
+At this point, publishers have the option of claiming ownership of their packages and publishing to ArtifactHub.io only.
+The CNB registry will read any new buildpacks that are published to ArtifactHub.io outside the CNB registry index
+publishing process because it is calling the ArtifactHub.io API.
+
+`pack` will continue reading from the CNB registry API.
+
+At this point, publishers can publish to either the buildpack registry index or ArtifactHub.io. By the end of 12 months,
+publishers should only use the ArtifactHub.io method for publishing, because the buildpack registry index will become
+readonly at this point and will not accept new publishes.
+
+### Requirement 6 - Update `pack` to pull from ArtifactHub.io
+
+We will introduce a new flag to `pack` called `registry-schema`. The options for this flag are `cnb` and `artifacthub`.
+This flag would be threaded through to all the current `pack` commands that rely on the registry, including all the
+`pack buildpack` commands. If this flag is set to `artifacthub`, `pack` will utilize ArtifactHub.io. For instance, the
+`pack buildpack register` command will call the ArtifactHub.io API.
+
+From month 0-12, the flag will default to `cnb` and users will be warned about the upcoming behavior changes.
+
+From month 12-18, the flag will default to `artifacthub`. On failures, users will be warned about the changes, and they
+will be able to set the flag to `cnb` for a period of time.
+
+After month 18, the flag will be deprecated and the only behavior will be the `artifacthub` schema behavior.
+
+### Requirement 7 - Terminate the buildpacks registry service
+
+From month 0-12, there will be only additive changes to the registry service. It will maintain full backwards
+compatibility for 12 months.
+
+From month 12-18, the registry service will be read-only. No buildpacks will be able to publish to the registry service
+after 12 months, but it will still serve `GET` requests so that `pack` does not break.
+
+After month 18, the registry service will be terminated. At this point, ArtifactHub.io will be the official source for
+buildpacks.
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+This is a disruptive change for buildpack publishers. The ownership and publishing model will be different.
+
+This is a destructive change that will affect `pack`. Any versions of `pack` that depend on the buildpack registry will
```

Comment:
this is the biggest risk IMO. we should think about how we might get ahead of this (like releasing something in `pack` ASAP that helps future proof)

Reply:
I don't know what the API for artifact hub is like or if this is in any way feasible, but what about running a proxy for requests from old versions of pack? If it could proxy and transparently map those requests to the new API, then old versions of pack would in theory keep working.

Not sure if there are any other platform implementations that integrated with the buildpacks registry, but that would be a way to keep other non-pack implementations working too.

Reply:
That's a really good thought @dmikusa I will look into this.